File: 283. Move Zeroes/moveZeroes.cpp
#include<iostream>
#include<vector>

using namespace std;

class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int left = 0;
        int sz = nums.size();

        for(int right = 1; right < sz; right++){

            if(nums[left] == 0){
                while(nums[right] == 0 && right < sz-1){
                    right++;
                }
                nums[left] = nums[right];
                nums[right] = 0;
            }
            left++;
        }
    }
};

File: 283. Move Zeroes/moveZeroes.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var moveZeroes = function (nums) {
    let left = 0;
    const len = nums.length

    for (let right = 1; right < len; right++) {

        if (nums[left] === 0) {

            while (
                right < len - 1 &&
                nums[right] === 0) {
                right++;
            }
            nums[left] = nums[right]
            nums[right] = 0
        }
        left++
    }
};